src/server/game/Cache/CharacterCache.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __CHARACTERCACHE_H
+#define __CHARACTERCACHE_H
+
+#include "Common.h"
+#include "ObjectGuid.h"
+
+#include <unordered_map>
+
+struct CharacterCacheEntry
+{
+    std::string Name;
+    uint32 AccountId;
+    uint8 Class;
+    uint8 Race;
+    uint8 Sex;
+    uint8 Level;
+    ObjectGuid::LowType GuildId;
+    uint32 ArenaTeamId[3];
+};
+
+typedef std::unordered_map<std::string, ObjectGuid> CharacterGuidByNameContainer;
```

Comment:
How about changing this map to <string, CharacterCacheEntry*> pointing inside the other container to avoid double lookups in *ByName methods? (and add Guid to CharacterCacheEntry)